File: lab_solutions/sol_lab2task1/src/publisher_node.cpp
#include "ros/ros.h"
//TODO: include the library for a msg for an integer
#include "std_msgs/Float64.h"

int main(int argc, char **argv)
{

    ros::init(argc, argv, "talker");

    //TODO: Define a nodehandle and a publisher.
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Float64>("/chatter", 1000);

    ros::Rate loop_rate(10);

    std_msgs::Float64 msg;
    msg.data = 1;
    int count = 0;

    while (ros::ok())
    {
	
	//TODO: Create a message and publish it. Your message should be a random number between 0 and the variable 'count'        
	if (count == 0){
		msg.data = 0;
	}
	else{	
		msg.data = rand() % count;
	}

        pub.publish(msg);

        ros::spinOnce();

        loop_rate.sleep();
        count++;
    }


    return 0;
}

File: lab_solutions/lab2task1/scripts/publisher_node.py
#!/usr/bin/env python

import rospy
import random
##TODO: include the library for a msg for an integer
from std_msgs.msg import Int64

def talker():
    ##TODO: Define a publisher for an integer
    pub = rospy.Publisher('chatter',Int64,queue_size=1000)
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(10) # 10hz
    count = 0
    while not rospy.is_shutdown():
        ##TODO: Create a message and publish it. Your message should be a random number between 0 and the variable 'count'
        data=random.randint(0,count)
        pub.publish(data)
        
        rate.sleep()
        count += 1

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
